### ISSUE TITLE
🐞 Bug report: sticky option causes website to jump when stopping scrolling

### ISSUE THREAD

**Issue by jormeer** [elementor/elementor#17767]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

With elementor pro 3.6 sticky top causes the screen to move when scrolling stops. I tested this with multiple websites.

Just sticky a element. Scroll down for a good amount en than a little bit back up. You will see the website will jump like 100px. This only happens in mobile phones. I used edge and chrome to test. Not possible to recreate on desktop.

### Steps to reproduce

Just sticky a element. Scroll down for a good amount en than a little bit back up. You will see the website will jump like 100px. This only happens in mobile phones. I used edge and chrome to test. Not possible to recreate on desktop.

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently using the steps above.

### System Info

					
== Server Environment ==
	Operating System: Linux
	Software: nginx/1.19.10
	MySQL version: mariadb.org binary distribution v10.5.9
	PHP Version: 8.1.2
	PHP Max Input Vars: 10000
	PHP Max Post Size: 128M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.9
	Site URL: https://lemonz.nl
	Home URL: https://lemonz.nl
	WP Multisite: No
	Max Upload Size: 128 MB
	Memory limit: 256M
	Permalink Structure: /blog/%postname%/
	Language: nl
	Timezone: 1
	Admin Email: contact@lemonz.nl
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor Child
	Version: 1.0.1
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 2.5.0
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: nl_NL
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.99 Safari/537.36

== Active Plugins ==
	Coming Soon Page, Maintenance Mode & Landing Pages by SeedProd
		Version: 6.9.0.8
		Author: SeedProd

	Ele Custom Skin
		Version: 3.1.4
		Author: Dudaster.com

	Elementor
		Version: 3.5.5
		Author: Elementor.com

	Elementor Pro
		Version: 3.6.0
		Author: Elementor.com

	ElementsKit Lite
		Version: 2.5.3
		Author: Wpmet

	ElementsKit Pro
		Version: 2.6.2
		Author: Wpmet

	SVG Support
		Version: 2.4.2
		Author: Benbodhi

== Must-Use Plugins ==
	Kinsta Must-use Plugins
		Version: 2.3.1
		Author: Kinsta Team

== Elementor experimenten ==
	Geoptimaliseerde DOM-uitvoer: Actief
	Verbeterd laden van assets: Actief
	Verbeterd laden van CSS: Actief
	Pictogrammen inline lettertype: Actief
	Toegang verbeteringen: Actief
	Import export template kit: Standaard actief
	Extra aangepaste onderbrekingspunten: Actief
	Verberg verouderde WordPress widgets voor zoekresultaten: Standaard actief
	Default to New Theme Builder: Standaard actief
	Hello Theme header & footer: Standaard inactief
	Landingspagina's: Actief
	Kleur sampler: Standaard actief
	Favoriete widgets: Standaard actief
	Beheerder bovenste balk: Actief
	Page Transitions: Standaard actief
	Form Submissions: Standaard actief
	Scroll Snap: Actief

== Log ==
	
Log: showing 20 of 362022-01-22 16:30:17 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.5.0-beta5',
  'to' => '3.5.3',
)]
2022-02-01 11:44:03 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.5.3',
  'to' => '3.5.4',
)]
2022-02-01 11:44:04 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.6.0-beta1',
  'to' => '3.6.0-beta4',
)]
2022-02-01 11:44:04 [info] elementor::elementor_updater Started 
2022-02-01 11:44:04 [info] Elementor/Upgrades - _on_each_version Start  
2022-02-01 11:44:04 [info] Elementor/Upgrades - _on_each_version Finished 
2022-02-01 11:44:04 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.5.3',
  'to' => '3.5.4',
)]
2022-02-01 11:44:04 [info] elementor-pro::elementor_pro_updater Started 
2022-02-01 11:44:04 [info] Elementor Pro/Upgrades - _on_each_version Start  
2022-02-01 11:44:04 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2022-02-01 11:44:04 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.6.0-beta1',
  'to' => '3.6.0-beta4',
)]
2022-02-01 13:32:41 [info] elementor-pro::elementor_pro_updater Started 
2022-02-01 13:32:41 [info] Elementor Pro/Upgrades - _on_each_version Start  
2022-02-01 13:32:41 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2022-02-01 13:32:41 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.6.0-beta4',
  'to' => '3.6.0',
)]
2022-02-04 16:16:19 [info] elementor::elementor_updater Started 
2022-02-04 16:16:19 [info] Elementor/Upgrades - _on_each_version Start  
2022-02-04 16:16:19 [info] Elementor/Upgrades - _on_each_version Finished 
2022-02-04 16:16:19 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.5.4',
  'to' => '3.5.5',
)]
2022-02-04 16:16:19 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.5.4',
  'to' => '3.5.5',
)]

PHP: showing 5 of 5PHP: 2022-02-01 11:41:37 [warning X 1][/www/lemonz_530/public/wp-content/plugins/elementor-pro/license/api.php::377] Undefined array key "features" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-02-03 10:17:27 [warning X 5][/www/lemonz_530/public/wp-content/plugins/elementor/core/common/modules/ajax/module.php::171] Undefined array key "data" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-02-03 11:03:02 [notice X 11443][/www/lemonz_530/public/wp-content/plugins/elementor/core/upgrade/manager.php::62] version_compare(): Passing null to parameter #2 ($version2) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-02-03 11:30:57 [notice X 1][/www/lemonz_530/public/wp-content/plugins/elementor-pro/modules/forms/submissions/database/entities/form-snapshot.php::70] Return type of ElementorPro\Modules\Forms\Submissions\Database\Entities\Form_Snapshot::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2022-02-04 16:16:19 [notice X 2][/www/lemonz_530/public/wp-content/plugins/elementor/core/logger/items/base.php::46] Return type of Elementor\Core\Logger\Items\Base::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 1 of 1JS: 2022-02-08 21:24:55 [error X 1][https://lemonz.nl/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.5.5:2:453832] Cannot read properties of null (reading \'getBoundingClientRect\') 

== Elementor - Compatibility Tag ==
	
	Ele Custom Skin: Compatibel
	Elementor Pro: Compatibel

== Elementor Pro - Compatibility Tag ==
	
	Ele Custom Skin: Niet compatibel

**Comment by abbasdv** [comment]
Same issue.
Just updated to Elementor 3.6.1 (2022-02-10) and it's still there. 

Sticky header causes the page to scroll down when resizing. 
Disabling sticky effect solves it. 
Theme: OceanWp 

**Same bug 2018 & being solved.**
https://github.com/elementor/elementor/issues/5740

**Comment by jormeer** [comment]
> Same issue. Just updated to Elementor 3.6.1 (2022-02-10) and it's still there.
> 
> Sticky header causes the page to scroll down when resizing. Disabling sticky effect solves it. Theme: OceanWp
> 
> **Same bug was mentioned and solved in 2018, see below:** #5740

I think that issue is something different. The issue 17767 is with elementor 3.6

**Comment by abbasdv** [comment]
Sorry, was late last night. I'll make a new post with this one.

**Comment by hettbeans** [comment]
Came here to open the same issue. This was driving me nuts before I realized I could duplicate it on a fresh install and it wasn't something I did.

**Comment by sstrantz** [comment]
I'm also experiencing this issue.  Have sticky headers on all sites and need them to remain that way.  Has anyone addressed this from Elementor yet?  @shilo-ey

**Comment by jormeer** [comment]
It also happens with the accordion widget. Just open it and the website jumps.

**Comment by andriussok** [comment]
@jormeer The problem is somewhere with the header parent container className (.elementor-location-header) height... even offsetHeight has its value, but page jumps equal to that height. Workaround - either set (.elementor-location-header) static height in px in CSS , or calculate child elements and set height with JS. See JS solution here [https://github.com/elementor/elementor/issues/17839#issuecomment-1041925736](url)

**Comment by 79ho3ein** [comment]
I have this problem too, even in the toggle widget.

**Comment by gmverdon** [comment]
I have this problem as well, this is an urgent problem!

**Comment by jormeer** [comment]
Still not fixed with latest update. ELementor really should get their game up. RIght now a website with a sticky element is useless.

**Comment by xxHeartFirexx** [comment]
> PLS fix i cant use 3.6 because all my sites use sticky headers

Yes indeed. Even after latest versions of plugin updates still the annoying scroll stop jump down. I have redesigned headers and turned sticky off to solve the issue. They really need to fix that asap.

**Comment by xxHeartFirexx** [comment]
Is there a fix for this coming out soon? Maybe next update?

**Comment by 79ho3ein** [comment]
I still experience this bug in the latest elementor and e pro versions. #18313

**Comment by ev88** [comment]
Also experiencing this.  Does not happen on initial page load when you haven't scrolled, only after you've scrolled a bit and activated the sticky header.  I can replicate it in Edge, Chrome, and Firefox.

Elementor Pro 3.6.4.

**Comment by 79ho3ein** [comment]
You only experience this bug on mobile or tablet, right? Everything works well on the desktop.

**Comment by 79ho3ein** [comment]
> > You only experience this bug on mobile or tablet, right? Everything works well on the desktop.
> 
> I am experiencing it on Desktop.

It is strange! Of course, I still use elementor pro 3.5.4.

**Comment by xxHeartFirexx** [comment]
Whitch brings me to this next question. Is the bug being worked on? Does anyone know if the elementor devs are aware of it and bringing in a fix in the next update? This is tbe question that I been wondering since Feb.

**Comment by jormeer** [comment]
Nothing reported that there is being worked on. Sometimes the devs are a joke here. This bug makes the new versions completely useless. I think almost all websites use sticky header. Why isn't this a priority bug for the team to fix in a week. And than I see the changelog of the recent patches. And I read those like oke nobody cares about this being fixed before this bug. Just howwwww. This is a ground breaking bug.

**Comment by 79ho3ein** [comment]
I'm a big fan of elementor, and working with elementor is one of my most enjoyable pastimes. Unfortunately, one of the most serious criticisms of the elementor team is that in addition to not paying attention to new user suggestions, they do not even pay attention to bug reports. I can report a long list of elementor bugs here that have been present in elementor since the beginning. But I know that it is not taken into account. It would be better for them to at least say that we are aware of this bug so that we can have some encouragement. But ...
I understand they are working on great features. They are the best and have made the most powerful page builder in the world. But it is not bad if they prioritize bug fixes.

**Comment by xxHeartFirexx** [comment]
I can totally understand that. Thanks for the info. I've come to a conclusion, im going to roll back pro to 3.5.4. Then just keep an eye out on future changelogs and see if its ever addressed.  Im still very surprised that this bug hasn't been taken care of being in the front end of the site where all visitors will get a nauseating scroll jump. Lets hope it will get fixed in the next pro update like the accordion and tab was. ( Dosen't make sense they fixed that, but not the sticky header scroll jump, ill never understand that)

**Comment by shilo-ey** [comment]
Thanks for reporting.

We will address this issue soon and release a patch.

Please follow this thread for updates.

Thanks!

**Comment by sstrantz** [comment]
Thank you @shilo-ey

**Comment by noelillinger** [comment]
Can only join this report with hope of a quick fix. It’s frustrating and affecting the customer surf experience enormously for weeks already. @shilo-ey - I tested it and can confirm, it only appears on android mobiles, mostly chrome. Safari and Chrome on iOS works fine, no jumping. 

Can you tell when the fix will be out?

**Comment by sstrantz** [comment]
On android Firefox also.

**Comment by ConsultMePL** [comment]
> Thanks for reporting.
> 
> We will address this issue soon and release a patch.
> 
> Please follow this thread for updates.
> 
> Thanks!

"Thanks for reporting, we will address this issue soon"

Seriously? You're like " ahh that's a minor bug, we'll take our time".

It's extremely critical issue that makes user experience very bad and looks awful. How come it's been 10 weeks since it was first reported and no fix yet?

**Comment by jormeer** [comment]
Yes this is really bad from elementor. I got a warning from kinsta that there is a security issue with elementor. And I have to update to newest version. BUT I can't because that will also introduce this bug. How can this not be priority

**Comment by hettbeans** [comment]
I can't believe this is still an issue. This is a tremendously amateurish bug to have present on a website in 2022 and it is a very bad look from Elementor.

**Comment by rayedgar** [comment]
Yes it's ridiculous this bug is still here. I cannot upgrade.

**Comment by noelillinger** [comment]
We need a QUICK FIX! 2022??

@shilo-ey , @iNewLegend , @Nevoss , @udidol 

Please tell us, when are we going to see this issue solved? It’s been around for too long, even customers contact us about this issue. 

A sticky header is a need nowadays, and its not working. Why do I feel like you don’t take this serious issue, really serious? 

Joining the feeling of many others I’ve been reading comments from in the web. 

Fix it asap..

**Comment by nirkons** [comment]
Bug still exists!! Is this going to be fixed?

**Comment by Michaelgimii** [comment]
I think there is a reason why elementor progresses very slowly. There is a reason why it takes months to fix every simple bug.
I think the elementor coding is old. Elementor was written 5 years ago. So there are problems with writing new code. So they spend most of their time adding new code without any problems and it takes a long time.
I do not know, maybe elementor needs to be rewritten. I'm really interested in knowing if I'm right or not @shilo-ey 
To be honest, we elementar users really prefer to spend 5 or 6 months rewriting elementor or even more. At least after that we have no problem adding or fixing bugs. Most importantly, it would be great to rewrite the entire elementor using react. Today, new page builders are written using react and have great results. Finally, I think if elementor does not want to lag behind its competitors, it definitely needs this rewrite. I do not know, maybe I'm wrong. But I hope a developer will answer me. thanks

**Comment by shilo-ey** [comment]
This bug is being handled and should be fixed in our next release (hotfix/Pro 3.7)

The reason for not updating here, is that we want to make sure the fix truly resolves the issue reported.

The issue caused by Chrome's browser sticky address bar makes these "jumps" and we don't want to compromise other browser behavior.

As for improving architecture and infrastructure - indeed that is something that we are focused on in our roadmap and are currently working on the matter. you can see that in our ongoing PRs and improvements "under the hood"

Thank you for your patience and an update will be shared here when the issue is resolved.

**Comment by sstrantz** [comment]
Thank you, @shilo-ey .

Just wanted to mention that I have the issue on Firefox mobile.  So it's not just a Chrome issue.

**Comment by ConsultMePL** [comment]
> This bug is being handled and should be fixed in our next release (hotfix/Pro 3.7)
> 
> The reason for not updating here, is that we want to make sure the fix truly resolves the issue reported.
> 
> The issue caused by Chrome's browser sticky address bar makes these "jumps" and we don't want to compromise other browser behavior.
> 
> As for improving architecture and infrastructure - indeed that is something that we are focused on in our roadmap and are currently working on the matter. you can see that in our ongoing PRs and improvements "under the hood"
> 
> Thank you for your patience and an update will be shared here when the issue is resolved.

But the issue isn't only in Chrome - it's in all browsers I've tested: Chrome, Samsung Internet Browser, Opera and Firefox.

The jump just happens in different moments...

**Comment by 79ho3ein** [comment]
This bug only exists in mobile browsers. I just did not see this bug in Safari browser. There is no problem on the desktop.

**Comment by 79ho3ein** [comment]
> This bug is being handled and should be fixed in our next release (hotfix/Pro 3.7)
> 
> The reason for not updating here, is that we want to make sure the fix truly resolves the issue reported.
> 
> The issue caused by Chrome's browser sticky address bar makes these "jumps" and we don't want to compromise other browser behavior.
> 
> As for improving architecture and infrastructure - indeed that is something that we are focused on in our roadmap and are currently working on the matter. you can see that in our ongoing PRs and improvements "under the hood"
> 
> Thank you for your patience and an update will be shared here when the issue is resolved.

thank you @shilo-ey , That means you even have a plan to overwrite the widgets in elementor?

**Comment by xxHeartFirexx** [comment]
3.7 is out. Is the scroll glitch fixed?

**Comment by ConsultMePL** [comment]
> 3.7 is out. Is the scroll glitch fixed?

Yeah, they finally fixed it in Beta 3

**Comment by hettbeans** [comment]
99% sure the issue is related to the viewport size being recalculated after the URL bar hides — definitely not endemic to Chrome exclusively but any mobile browser with that animated UX.

**Comment by avivu** [comment]
Hello,
This issue has been resolved in Elementor Pro v3.7.0

Feel free to update

Thanks!

**Comment by ConsultMePL** [comment]
@avivu why the issue is closed if it's been back for a month?! Why the fu** are you guys removing my comments regarding this issue? What the hell I'm paying $500 for?

**Comment by 79ho3ein** [comment]
Hi @MinneapolisMade , It would be great if you could report this issue in a [new issue](https://github.com/elementor/elementor/issues/new?assignees=&labels=status%2Fawaiting_triage&template=bug.yml) so developers can track it down faster. **(Please share the bug report with reproduction steps and also the system info)**
Unfortunately, I don't have an iPhone and I can't test this issue. ...
thank you
